src/frontend/src/pages/PIAForm/index.tsx

```diff
@@ -383,6 +388,17 @@ const PIAFormPage = () => {
         } else {
           navigate(-1);
         }
+      } else if (buttonValue === 'save') {
+        const newPia = await upsertAndUpdatePia();
+        if (newPia?.id) {
+          navigate(
+            buildDynamicPath(routes.PIA_VIEW, {
+              id: pia.id,
+            }),
+          );
+        } else {
+          navigate(-1);
```

Comment:
why do we need to navigate back when "new" ??
We can disable the button when we don't have the id.

Reply:
remove this else statement.